src/sentry/static/sentry/app/views/alerts/rules/details/metricChart.tsx

```diff
@@ -1,145 +1,250 @@
-  const series: LineChartSeries[] = [...data];
-  // Ensure series data appears above incident lines
-  series[0].z = 100;
-  const dataArr = data[0].data;
-  const maxSeriesValue = dataArr.reduce(
-    (currMax, coord) => Math.max(currMax, coord.value),
-    0
-  );
-  const firstPoint = Number(dataArr[0].name);
-  const lastPoint = dataArr[dataArr.length - 1].name;
-  const resolvedArea = {
-    seriesName: 'Resolved Area',
-    type: 'line',
-    markLine: MarkLine({
-      silent: true,
-      lineStyle: {color: theme.green300, type: 'solid', width: 4},
-      data: [[{coord: [firstPoint, 0]}, {coord: [lastPoint, 0]}] as any],
-    }),
-    data: [],
+export default class MetricChart extends React.PureComponent<Props, State> {
+  state = {
+    width: -1,
+    height: -1,
   };
-  series.push(resolvedArea);
-  if (incidents) {
-    // select incidents that fall within the graph range
-    const periodStart = moment.utc(firstPoint);
-    const filteredIncidents = incidents.filter(incident => {
-      return !incident.dateClosed || moment(incident.dateClosed).isAfter(periodStart);
-    });
-
-    const criticalLines = filteredIncidents.map(incident => {
-      const detectTime = Math.max(moment(incident.dateStarted).valueOf(), firstPoint);
-      let resolveTime;
-      if (incident.dateClosed) {
-        resolveTime = moment(incident.dateClosed).valueOf();
-      } else {
-        resolveTime = lastPoint;
-      }
-      const line = [{coord: [detectTime, 0]}, {coord: [resolveTime, 0]}];
-      return line;
-    });
-    const criticalArea = {
-      seriesName: 'Critical Area',
-      type: 'line',
-      markLine: MarkLine({
-        silent: true,
-        lineStyle: {color: theme.red300, type: 'solid', width: 4},
-        data: criticalLines as any,
-      }),
-      data: [],
-    };
-    series.push(criticalArea);
-
-    const incidentValueMap: Record<number, string> = {};
-    const incidentLines = filteredIncidents.map(({dateStarted, identifier}) => {
-      const incidentStart = moment(dateStarted).valueOf();
-      incidentValueMap[incidentStart] = identifier;
-      return {xAxis: incidentStart};
-    });
-    const incidentLinesSeries = {
-      seriesName: 'Incident Line',
+
+  ref: null | ReactEchartsRef = null;
+
+  /**
+   * Syncs component state with the chart's width/heights
+   */
+  updateDimensions = () => {
+    const chartRef = this.ref?.getEchartsInstance?.();
+    if (!chartRef) {
+      return;
+    }
+
+    const width = chartRef.getWidth();
+    const height = chartRef.getHeight();
+    if (width !== this.state.width || height !== this.state.height) {
+      this.setState({
+        width,
+        height,
+      });
+    }
+  };
+
+  handleRef = (ref: ReactEchartsRef): void => {
+    // When chart initially renders, we want to update state with its width, as well as initialize starting
+    // locations (on y axis) for the draggable lines
```

Comment:
probably can edit this comment since there are no draggable lines

Reply:
👍